src/error.rs

```diff
@@ -0,0 +1,31 @@
+use std::fmt::Debug;
+
+pub enum VMError {
```

Comment:
All the variants in this Error typo are just String wrappers. From a developer point of view this is not really useful since there is not a lot that can be done with that String (only showing it to the end user). 

If they are only used for user-facing purposes (showing them in the terminal) that's fine but if you are developing a library you need to add more data that can be used by the developer who is interacting with this module and needs to handle this error.